app/models/casebook/resource.rb

```diff
@@ -0,0 +1,32 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Resource < Casebook::Contents
+  default_scope {where.not(material_id: nil)}
+
+  belongs_to :material, polymorphic: true, inverse_of: :casebooks, required: true
```

Comment:
The benefit of breaking things up like this (rather than pointing directly to a Case) is that you can clone a casebook without needing to copy all of its resources, just copying the references. Then the actual resource can be copied when the user actually edits it (copy-on-write). This will make cloning faster for large casebooks,  as well as less data-heavy for those where few changes are made to the clone.